### PR TITLE
use jwt instead of basicAuth for webEnv

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
use jwt instead of basicAuth for webEnv

## Motivation and Context
Just us a different Authorization scheme

## How to test this PR?
As per #10215 you need a webhook server to validate
the JWT token. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :x: [more...](#f05c0c52-9cfa-4eee-bc4c-cd8c9a28d239) |
| mint-gateway-azure.sh | :x: [more...](#84874d46-183c-40f8-9349-194df9e2999c) |
<a name="f05c0c52-9cfa-4eee-bc4c-cd8c9a28d239"></a>
### 10246-3539310/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31591
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 25e31f4d8668:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 1 minutes and 7 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 31 seconds
(8/15) Running minio-dotnet tests ... done in 28 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 20 seconds
{
  "args": {
    "destination": {
      "Bucket": "minio-go-test-h12nltko51zhef9m",
      "Object": "srcObject",
      "Encryption": {},
      "UserMetadata": null,
      "ReplaceMetadata": false,
      "UserTags": null,
      "ReplaceTags": false,
      "LegalHold": "",
      "Mode": "",
      "RetainUntilDate": "0001-01-01T00:00:00Z",
      "Size": 0,
      "Progress": null
    },
    "source": {
      "Bucket": "minio-go-test-h12nltko51zhef9m",
      "Object": "srcObject",
      "VersionID": "",
      "MatchETag": "",
      "NoMatchETag": "",
      "MatchModifiedSince": "0001-01-01T00:00:00Z",
      "MatchUnmodifiedSince": "0001-01-01T00:00:00Z",
      "MatchRange": false,
      "Start": 0,
      "End": 0,
      "Encryption": null
    }
  },
  "duration": 322,
  "error": "This copy request is illegal because it is trying to copy an object to itself without changing the object's metadata, storage class, website redirect location or encryption attributes.",
  "function": "CopyObject(destination, source)",
  "message": "CopyObject failed",
  "name": "minio-go: testEncryptedSSES3ToSSES3CopyObject",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 20 seconds
(10/15) Running minio-js tests ... done in 34 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 21 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 1 seconds
(13/15) Running s3select tests ... done in 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="84874d46-183c-40f8-9349-194df9e2999c"></a>
### 10246-3539310/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31042
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 629ca8336583:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 18 seconds
(4/15) Running aws-sdk-ruby tests ... done in 17 seconds
(5/15) Running awscli tests ... done in 2 minutes and 36 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 56 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 33 seconds
(9/15) Running minio-go tests ... FAILED in 5 minutes and 26 seconds
{
  "args": {},
  "duration": 947,
  "error": "The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.",
  "function": "CopyObjectPart(destination, source)",
  "message": "CompleteMultipartUpload call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 5 minutes and 55 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 34 seconds
(11/15) Running minio-py tests ... FAILED in 1 seconds
{
  "name": "minio-py:test_negative_make_bucket_invalid_name",
  "status": "FAIL",
  "args": {
    "location": "default value ('us-east-1')",
    "bucket_name": ".minio-py-test-7c8630a6-1687-4724-8cb7-1f525864561d"
  },
  "message": "NoSuchBucket: message: The specified bucket does not exist.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 241, in test_negative_make_bucket_invalid_name\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 130, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 335, in make_bucket\n    raise ResponseError(response, method, bucket_name).get_exception()\nminio.error.NoSuchBucket: NoSuchBucket: message: The specified bucket does not exist.\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 443
}
(11/15) Running s3cmd tests ... done in 1 minutes and 20 seconds
(12/15) Running s3select tests ... done in 1 minutes and 2 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Review by kannappanr — APPROVED** [review]
LGTM

File: pkg/env/web_env.go [hunk 1]
```diff
@@ -96,10 +98,20 @@ func fetchEnvHTTP(envKey string, u *url.URL) (string, error) {
...
-	if username != "" && password != "" {
-		req.SetBasicAuth(username, password)
+	claims := &jwt.StandardClaims{
+		ExpiresAt: int64(15 * time.Minute),
+		Issuer:    username,
+		Subject:   envKey,
+	}
+
+	token := jwt.NewWithClaims(jwt.SigningMethodHS512, claims)
+	ss, err := token.SignedString([]byte(password))
```
Comment (kerneltime) [inline pkg/env/web_env.go:108]: Operator will need to have read access to the password and it needs to be per minio instance but that is not relevant here

**Review by kerneltime — APPROVED** [review]
Nit: We should move the logic to extract username and password into its own routine.

File: pkg/env/web_env.go (hunk 1 above)
Comment (harshavardhana) [reply pkg/env/web_env.go:108]: This secret is actually operator webhook secret not the tenant secret, so even if lost or compromised can be regenerated.

**Pull request merged**